pkg/vrct/vrctFs/revert.go

```diff
@@ -0,0 +1,121 @@
+package vrctFs
+
+import (
+	"fmt"
+	"os"
+)
+
+const revertTmpPath = "/tmp/spito-vrct/fs-revert"
+
+const (
+	removeFile = iota
+	removeDirAll
+	replaceContent
+)
+
+type RevertStep struct {
+	path   string
+	action int
+	// oldContentPath field is optional
+	oldContentPath *string
+}
+
+type RevertSteps struct {
+	steps         []RevertStep
+	revertTempDir string
+}
+
+func NewRevertSteps() (RevertSteps, error) {
+	err := os.MkdirAll(revertTmpPath, os.ModePerm)
+	if err != nil {
+		return RevertSteps{}, err
+	}
+
+	dir, err := os.MkdirTemp(revertTmpPath, "")
+	if err != nil {
+		return RevertSteps{}, err
+	}
+
+	return RevertSteps{
+		revertTempDir: dir,
+	}, nil
+}
+
+func (r *RevertSteps) RemoveFile(path string) {
+	r.steps = append(r.steps, RevertStep{
+		path:   path,
+		action: removeFile,
+	})
+}
+
+func (r *RevertSteps) RemoveDirAll(path string) {
+	r.steps = append(r.steps, RevertStep{
+		path:   path,
+		action: removeDirAll,
+	})
+}
+
+func (r *RevertSteps) BackupOldContent(path string) error {
+	oldFile, err := os.ReadFile(path)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	tempContentFile, err := os.CreateTemp(r.revertTempDir, "old-")
```

Comment:
I suggest something like `~/.local/state`

Reply:
But this is temporary directory which is going to be deleted later

Reply:
Where is backup stored?